include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_utils.h

```diff
@@ -769,6 +770,15 @@ class __static_monotonic_dispatcher<::std::integer_sequence<::std::uint16_t, _X,
     }
 };
 
+template <typename _ExecutionPolicy, typename _SizeType>
+bool
+__supports_sub_group_size(const _ExecutionPolicy& __exec, _SizeType __target_size)
```

Comment:
Since the only usage of `__target_size` in this function casts it to `std::size_t`, would it make sense to take the parameter by `std::size_t` directly instead of using a template?
```suggestion
__supports_sub_group_size(const _ExecutionPolicy& __exec, std::size_t __target_size)
```

Reply:
Good point. I've added this and removed the `static_cast` below.